File: Organisms Game/sowthistle.cpp
#include "sowthistle.h"
#include "colors.h"

Sow::Sow(World *world, Point coord, int bornRound)
:Plant(OrganismType::GRASS,world,coord,SOW_STRENGTH,SOW_INITIATIVE,bornRound)
{
    this->colour=GREEN;
    this->symbol=SOW_SYMBOL;
}

void Sow::Action()
{
    for(int i=0;i<3;i++)
    {
        if(rand()%100>75) Sowing();
    }
}

string Sow::OrganismTypeToString()
{
	return "Sow Thistle";
}

File: Java Organisms Game/ProjectWorld/Sowthistle.java
package ProjectWorld;

/**
 *
 * @author kprze
 */
public class Sowthistle extends Plant{
    public Sowthistle(Coords coords, World world) {
        super(coords, world);
        this.name="Sow Thistle";
    }
    
    @Override
    public void action()
    {
        multiply();
        multiply();
        multiply();
    }
}
